src/MessagePack.SourceGenerator/Utils/ImmutableEquatableArray.cs

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) All contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma warning disable SA1127 // Generic type constraints must be on own line
+#pragma warning disable SA1402 // File may only contain a single type
+#pragma warning disable SA1649 // File name should match first type name
+
+using System.Collections;
+
+namespace MessagePack.SourceGenerator;
+
+/// <summary>
+/// Provides an immutable list implementation which implements sequence equality.
+/// </summary>
+public sealed class ImmutableEquatableArray<T> : IEquatable<ImmutableEquatableArray<T>>, IReadOnlyList<T>
```

Comment:
This looks good, and it must have been quite a bit of work.
Did you consider just sticking with `ImmutableArray<T>` in `GenericTypeParameterInfo` and simply overriding its `Equals` method to know how to do by-value comparison?

Reply:
Actually I copied the code from [dotnet-runtime](https://github.com/dotnet/runtime/blob/main/src/libraries/Common/src/SourceGenerators/ImmutableEquatableArray.cs). 

If `Equals` is implemented manually, when a property is added, it might be forgetten to be added to `Equals`.  So I prefer to avoid writing it manually.